punchcut/src/source/glyf/deltas.rs

```diff
@@ -0,0 +1,289 @@
+fn interpolate_deltas(
+    points: &[Point<i32>],
+    flags: &[PointFlags],
+    contours: &[u16],
+    out_points: &mut [Point<Fixed>],
+) -> Option<()> {
+    if contours.is_empty() {
+        return Some(());
+    }
+    let mut point_ix = 0usize;
+    for &end_point_ix in contours {
+        let end_point_ix = end_point_ix as usize;
+        let first_point_ix = point_ix;
+        // Search for first point that has a delta.
+        while point_ix <= end_point_ix && !flags.get(point_ix)?.has_marker(PointMarker::HAS_DELTA) {
+            point_ix += 1;
+        }
+        if point_ix <= end_point_ix {
+            let first_delta_ix = point_ix;
+            let mut cur_delta_ix = point_ix;
+            point_ix += 1;
+            // Search for next point that has a delta...
+            while point_ix <= end_point_ix {
+                if flags.get(point_ix)?.has_marker(PointMarker::HAS_DELTA) {
+                    // ... and interpolate intermediate points.
+                    interpolate_range(
+                        points,
+                        cur_delta_ix + 1,
+                        point_ix - 1,
+                        cur_delta_ix,
+                        point_ix,
+                        out_points,
+                    );
+                    cur_delta_ix = point_ix;
+                }
+                point_ix += 1;
+            }
+            // If we only have a single delta, shift the contour.
+            if cur_delta_ix == first_delta_ix {
+                shift_range(
+                    points,
+                    first_point_ix,
+                    end_point_ix,
+                    cur_delta_ix,
+                    out_points,
+                )?;
+            } else {
+                // Otherwise, handle remaining points at beginning and end of contour.
+                interpolate_range(
+                    points,
+                    cur_delta_ix + 1,
+                    end_point_ix,
+                    cur_delta_ix,
+                    first_delta_ix,
+                    out_points,
+                )?;
+                if first_delta_ix > 0 {
+                    interpolate_range(
+                        points,
+                        first_point_ix,
+                        first_delta_ix - 1,
+                        cur_delta_ix,
+                        first_delta_ix,
+                        out_points,
+                    )?;
+                }
+            }
+        }
+    }
+    Some(())
+}
+
+/// Shift the coordinates of all points between `start` and `end` inclusive
+/// using the difference given by the point at `ref_`.
+///
+/// Modeled after the FreeType implementation: <https://github.com/freetype/freetype/blob/bbfcd79eacb4985d4b68783565f4b494aa64516b/src/truetype/ttgxvar.c#L3776>
+fn shift_range(
+    points: &[Point<i32>],
+    start: usize,
+    end: usize,
+    ref_: usize,
```

Comment:
Why the trailing _?

Reply:
`ref` is a keyword. I changed these to `ref_ix` to copy the style in other parts of the code and to avoid the dangling underscore.